substrate/primitives/io/src/lib.rs

```diff
@@ -2712,55 +2711,289 @@ pub trait Crypto {
 	///
 	/// The `seed` needs to be a valid utf8.
 	///
-	/// Returns the public key.
-	#[cfg(feature = "bls-experimental")]
+	/// Stores the public key in the provided output buffer.
+	#[wrapped]
 	fn bls381_generate(
 		&mut self,
 		id: PassPointerAndReadCopy<KeyTypeId, 4>,
 		seed: PassFatPointerAndDecode<Option<Vec<u8>>>,
-	) -> AllocateAndReturnPointer<bls381::Public, 144> {
+		out: PassPointerAndWrite<&mut bls381::Public, 144>,
```

Comment:
Why did you choose to pass the output buffer instead of returning it by value?
It may be functionally equivalent, but this API design differs from the other "generate" methods, which return a value. I would prefer a consistent design.